src/gsy_e/models/area/scm_manager.py

```diff
@@ -0,0 +1,249 @@
+
+    def add_home_data(self, home_uuid: str, home_name: str,
+                      grid_fees_eur: float, coefficient_percent: float,
+                      market_maker_rate_eur: float, feed_in_tariff_eur: float,
+                      production_kWh: float, consumption_kWh: float):
+        # pylint: disable=too-many-arguments
+        """Import data for one individual home."""
+        if grid_fees_eur is None:
+            grid_fees_eur = 0.0
+        self._home_data[home_uuid] = HomeAfterMeterData(
+            home_uuid, home_name,
+            grid_fees_eur=grid_fees_eur,
+            sharing_coefficient_percent=coefficient_percent,
+            market_maker_rate_eur=market_maker_rate_eur,
+            feed_in_tariff_eur=feed_in_tariff_eur,
+            production_kWh=production_kWh, consumption_kWh=consumption_kWh)
+
+    def calculate_community_after_meter_data(self):
+        """Calculate community data by aggregating all single home data."""
+        self.community_data.production_kWh = sum(
+            data.production_kWh for data in self._home_data.values())
+        self.community_data.consumption_kWh = sum(
+            data.consumption_kWh for data in self._home_data.values())
+        self.community_data.self_consumed_energy_kWh = sum(
+            data.self_consumed_energy_kWh for data in self._home_data.values())
+        self.community_data.energy_surplus_kWh = sum(
+            data.energy_surplus_kWh for data in self._home_data.values())
+        self.community_data.energy_need_kWh = sum(
+            data.energy_need_kWh for data in self._home_data.values())
+
+        for home_data in self._home_data.values():
+            home_data.set_community_production(self.community_data.energy_surplus_kWh)
+
+        self.community_data.energy_bought_from_community_kWh = sum(
+            data.energy_bought_from_community_kWh for data in self._home_data.values())
+        self.community_data.energy_sold_to_grid_kWh = sum(
+            data.energy_sold_to_grid_kWh for data in self._home_data.values())
+
+    def calculate_home_energy_bills(
+            self, home_uuid: str) -> None:
+        """Calculate energy bills for one home."""
+        assert home_uuid in self._home_data
+
+        home_data = self._home_data[home_uuid]
+
+        market_maker_rate_eur = home_data.market_maker_rate_eur
+        grid_fees_eur = home_data.grid_fees_eur
+        feed_in_tariff_eur = home_data.feed_in_tariff_eur
+
+        market_maker_rate_decreased_fees = (
+                market_maker_rate_eur + grid_fees_eur * self._grid_fees_reduction)
+        market_maker_rate_normal_fees = market_maker_rate_eur + grid_fees_eur
+
+        base_energy_bill = (
+                home_data.energy_need_kWh * market_maker_rate_normal_fees -
+                home_data.energy_surplus_kWh * feed_in_tariff_eur)
+
+        if home_data.allocated_community_energy_kWh > home_data.energy_need_kWh:
+            if home_data.energy_surplus_kWh > 0.0:
```

Comment:
I am confused here. How can the home have a surplus `home_data.energy_surplus_kWh > 0.0` and also need energy `home_data.allocated_community_energy_kWh > 0`?

Reply:
This is a tricky one indeed. The energy need of the house is the variable energy_need_kWh. The allocated energy to a house is not related to this though. Essentially, the house can be allocated more energy than it needs, hence the case that you describe: the house has energy surplus, but at the same time the community has allocated some energy to it.

Reply:
Ah! Thanks! This made me understand the whole if block here. A inline comment would be helpful. But you already promised to move parts of the logic into sub methods right? This will work too. Thanks!

Reply:
Yes, I will move it to individual methods (have more or less done it in GSYE-258 branch) and ping you in the next PR, thanks!